client/client_test.go

```diff
@@ -264,34 +265,90 @@ func TestChecksumMatch(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestSizeMismatchLong(t *testing.T) {
```

Comment:
@diogomonica tests for size mismatches. I realized while writing them that size is treated (correctly) as an upper bound, so that actual errors are caught elsewhere for other reasons.